tests/gen_code.c

```diff
@@ -0,0 +1,302 @@
+/*
+ * libiio - Library for interfacing industrial I/O (IIO) devices
+ *
+ * Copyright (C) 2014, 2019 Analog Devices, Inc.
+ * Author: Paul Cercueil <paul.cercueil@analog.com>
+ *         Robin Getz <robin.getz@analog.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * */
+
+#include <stdio.h>
+#include <string.h>
+#include <errno.h>
+#include <iio.h>
+//#include "gen_code.h"
+
+static FILE *fd = NULL;
+static char *uri;
+static enum languages {
+	C_LANG,
+	PYTHON_LANG,
+	UNSUPPORTED_LANG,
+} lang = UNSUPPORTED_LANG;
+
+bool gen_test_path(const char *gen_file)
+{
+	FILE *test;
+	char *first, *last;
+
+	if (!gen_file)
+		return false;
+
+	if (gen_file[0] == '-')
+		return false;
+
+	last = first = strchr(gen_file, '.');
+	if (!first)
+		return false;
+
+	while ((first = strchr(last, '.')))
+		last = first + 1;
+
+	if (!strcmp(last, "c"))
+		lang = C_LANG;
+	else if (!strcmp(last, "py"))
+		lang = PYTHON_LANG;
+	else {
+		fprintf(stderr, "Not a 'c' or 'py' file\n");
+		return false;
+	}
+
+	test = fopen(gen_file, "w");
+	if (!test)
+		return false;
+	fclose(test);
+
+	return true;
+}
+
+void gen_start(const char *gen_file)
+{
+	if (!gen_file)
+		return;
+
+	if (lang == UNSUPPORTED_LANG)
+		return;
+
+	fd = fopen(gen_file, "w");
+	if (!fd) {
+		printf("Error '%s' opening file: %s\n", strerror(errno),gen_file);
```

Comment:
The message could also go to the 'stderr' stream.

Reply:
fixed